### PR TITLE
Minor CI tweaks

### PR DESCRIPTION
**Pull request by andresilva**
Assorted tweaks that I applied to my own fork while CI was failing in this repo. Feel free to pick whatever you think it's worth it.

### CODE REVIEW

File: .github/workflows/test_dep_update.sh [hunk 1]
```diff
@@ -7,9 +7,17 @@ git branch depupdate
 git checkout depupdate
 git reset --hard origin/master
 json="$(gh pr list --search "author:app/github-actions" --json headRefName,number)"
+if [[ $json == "[]" ]]; then
+    echo "No Dependency update PRs to merge"
+    exit
+fi
```
Comment (andresilva) [inline .github/workflows/test_dep_update.sh:13]: If there's no updates then the CI action will fail without this, which is more common if we reduce the schedule from 24 hours.

File: .github/workflows/test_dep_update.sh [hunk 2]
```diff
@@ -7,9 +7,17 @@ git branch depupdate
...
 echo "$json" | jq '.[] | .headRefName | @text' | xargs -L1 -- git pull origin
 if nix build .#checks.x86_64-linux.init-example-el; then
-    echo "$json" | jq ".[] | .number | @text" | xargs -L1 -- gh pr merge --squash --delete-branch
+    prs=$(echo "$json" | jq ".[] | .number")
+    for pr in $prs; do
+        sleep 5
+        gh pr merge --squash --delete-branch "$pr"
+    done
```
Comment (andresilva) [inline .github/workflows/test_dep_update.sh:20]: Whenever there's a bunch of PRs to merge at the same time the GitHub API needs some time to update its internal state before allowing another PR to be merged.

File: .github/workflows/update-dependencies.yml [hunk 3]
```diff
@@ -2,7 +2,7 @@ name: "Update Dependencies"
 on:
   workflow_dispatch:
   schedule:
-    - cron: '0 0 * * FRI'
+    - cron: '0 */6 * * *'
```
Comment (andresilva) [inline .github/workflows/update-dependencies.yml:5]: Why not? :)

**Comment by ckiee** [comment]
Thanks a lot. I've been poking it with little sticks every time I got an email for `test_dep_update` running again (and usually failing) but haven't had the motivation to actually debug it properly because of how long it takes to test each change, so this is very nice to have. 

(Also, it'd be nice if [some of] the review comments were inline for future reference)

File: .github/workflows/update-dependencies.yml (hunk 3 above)
Comment (ckiee) [reply .github/workflows/update-dependencies.yml:5]: I have GH set to notify about all activities and this would be pretty spammy

**Comment by ckiee** [comment]
Added the comments, dropping the "every 6 hours" bit and force pushing again, but you can find the current state of your branch in my `ckiee/nix-doom-emacs`'s `ci-tweaks-before` branch.

**Pull request merged**